bar.m

```diff
@@ -38,76 +38,30 @@
         Method     swizzledMethod         = class_getInstanceMethod(targetClass, swizzledSelector) ;
         IMP        swizzledImplementation = method_getImplementation(swizzledMethod) ;
         const char *swizzledEncoding      = method_getTypeEncoding(swizzledMethod) ;
 
-        // will likely be false because originalSelector already exists, but just in case
-        // it goes away in a future macOS update (it is undocumented, after all) lets not
-        // crash when method_exchangeImplementations is invoked. True, this will end up
-        // having no effect then, but we're no worse off than before...
-        BOOL didAddMethod = class_addMethod(
-                                               targetClass,
-                                               originalSelector,
-                                               swizzledImplementation,
-                                               swizzledEncoding
-                                           ) ;
-
-        if (didAddMethod) {
-            class_replaceMethod(
-                targetClass,
-                swizzledSelector,
-                originalImplementation,
-                originalEncoding
-            ) ;
-            [LuaSkin logDebug:[NSString stringWithFormat:@"_crashOnException and xxx_hammerspoon_crashOnException added to %s (probably means this fix ineffective)", class_getName(targetClass)]] ;
-        } else {
-            method_exchangeImplementations(originalMethod, swizzledMethod);
-            [LuaSkin logDebug:[NSString stringWithFormat:@"_crashOnException and xxx_hammerspoon_crashOnException swizzled in %s", class_getName(targetClass)]] ;
-        }
+        method_exchangeImplementations(originalMethod, swizzledMethod);
+        [LuaSkin logWarn:[NSString stringWithFormat:@"markActiveFunctionRowsAsDimmed and xxx_hammerspoon_markActiveFunctionRowsAsDimmed swizzled in %s", class_getName(targetClass)]] ;
     });
 }
 
--(void) xxx_hammerspoon_crashOnException:(NSException *)exception {
-    if ([exception.debugDescription rangeOfString:@"NSFunctionRow"].location != NSNotFound) {
-        NSLog(@"_crashOnException override: %@", exception.debugDescription) ;
-        // may remove this later, but for now I want to be able to see when this occurs in the HS
-        // console and not just when I happen to be running the Console application
-        [LuaSkin logWarn:[NSString stringWithFormat:@"%s._crashOnException override: %@", USERDATA_TAG, exception.debugDescription]] ;
-        return ;
-    } else {
-        // note that since we "swizzled" _crashOnException, this name now points to the
-        // priginal version
-        [self xxx_hammerspoon_crashOnException:exception] ;
-    }
+-(void) xxx_hammerspoon_markActiveFunctionRowsAsDimmed:(BOOL)dimmed {
+    [LuaSkin logWarn:[NSString stringWithFormat:@"%s.markActiveFunctionRowsAsDimmed Triggered: %@", USERDATA_TAG, dimmed]] ;
```

Comment:
depending upon what the function actually does, you may need to conditionally invoke the "original" (now swizzled) method via `[self xxx_hammerspoon_markActiveFunctionRowsAsDimmed:dimmed] ;` here -- and yes, this is the correct name because we swizzled it out.

Reply:
Looking here:

https://github.com/pock/pock/blob/e04717699d39075e4ed586c93337ad9032225811/Pock/Private/TouchBarHelper.swift#L184

It looks like Pock is just using it to "tear down" their Touch Bar when "dimmed" and vice-versa. I'm assuming that if we invoke the "original" method, we'll just get a crash?

Reply:
Which is why I said "may" and "conditionally"